src/box/xrow.c

```diff
@@ -1798,19 +1806,18 @@ xrow_decode_subscribe(const struct xrow_header *row,
 		xrow_on_decode_err(row, ER_INVALID_MSGPACK, "request body");
 		return -1;
 	}
-
-	if (replicaset_uuid != NULL)
-		*replicaset_uuid = uuid_nil;
-	if (instance_uuid != NULL)
-		*instance_uuid = uuid_nil;
-	if (vclock != NULL)
-		vclock_create(vclock);
-	if (version_id != NULL)
-		*version_id = 0;
-	if (anon != NULL)
-		*anon = false;
-	if (id_filter != NULL)
-		*id_filter = 0;
+	if (req->replicaset_uuid != NULL)
+		*req->replicaset_uuid = uuid_nil;
+	if (req->instance_uuid != NULL)
+		*req->instance_uuid = uuid_nil;
+	if (req->vclock != NULL)
+		vclock_create(req->vclock);
+	if (req->version_id != NULL)
+		*req->version_id = 0;
+	if (req->is_anon != NULL)
+		*req->is_anon = false;
+	if (req->id_filter != NULL)
+		*req->id_filter = 0;
```

Comment:
If you apply my suggestion with memsetting the `smth_request` in `xrow_decode_smth`, you won't have to nullify the fields here explicitly.

Reply:
It works for now because `uuid_nil` and `vclock_create` are zeros. I removed these initializations for now, but in future it might complicate things.

Reply:
> It works for now because uuid_nil and vclock_create are zeros. I removed these initializations for now, but in future it might complicate things.

Yep. Let's try to keep things this way then.